libmscore/mscore.cpp

```diff
@@ -164,6 +164,7 @@ void Direction::fillComboBox(QComboBox* cb)
       }
 
 static Spatium doubleToSpatium(double d)       { return Spatium(d); }
+static Direction intToDirection(int i)         { return Direction(i); }
```

Comment:
I now realize the problem is the way undoChangeProperty() is called three times in Score::cmdFlip() explicitly converting Direction to int:
<pre><code>1640:            undoChangeProperty(chord, P_ID::STEM_DIRECTION, int(dir));
1683:            undoChangeProperty(a, P_ID::DIRECTION, int(d));
1689:            undoChangeProperty(tuplet, P_ID::DIRECTION, int(d));
</code></pre>
There are other calls to undoChangeProperty, that don't convert Direction to int, for example:
<pre><code>1647:            undoChangeProperty(beam, P_ID::STEM_DIRECTION, dir);</code></pre>
There are calls to undoChangeProperty that convert: Hairpin::Type to int and Element::Placement to int. On both of these types Q_DECLARE_METATYPE is used.

There are calls to undoChangeProperty for TimeSigType, on which Q_DECLARE_METATYPE is (can) not be used.